src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -96,26 +96,20 @@ private static bool FirstCharInUInt32IsAscii(uint value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static unsafe nuint GetIndexOfFirstNonAsciiByte(byte* pBuffer, nuint bufferLength)
         {
-            // If SSE2 is supported, use those specific intrinsics instead of the generic vectorized
-            // code below. This has two benefits: (a) we can take advantage of specific instructions like
-            // pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
+            // If 256/512-bit aren't supported but SSE2 is supported, use those specific intrinsics instead of
+            // the generic vectorized code. This has two benefits: (a) we can take advantage of specific instructions
+            // like pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
             // this method is running.
 
-            if (Vector512.IsHardwareAccelerated || Vector256.IsHardwareAccelerated)
-            {
-                return GetIndexOfFirstNonAsciiByte_Vector(pBuffer, bufferLength);
-            }
-            else if (Sse2.IsSupported || (AdvSimd.IsSupported && BitConverter.IsLittleEndian))
+            if (!Vector512.IsHardwareAccelerated &&
+                !Vector256.IsHardwareAccelerated &&
+                (Sse2.IsSupported || (AdvSimd.IsSupported && BitConverter.IsLittleEndian)))
             {
                 return GetIndexOfFirstNonAsciiByte_Intrinsified(pBuffer, bufferLength);
             }
-            else if (Vector128.IsHardwareAccelerated)
-            {
-                return GetIndexOfFirstNonAsciiByte_Vector(pBuffer, bufferLength);
-            }
             else
             {
-                return GetIndexOfFirstNonAsciiByte_Default(pBuffer, bufferLength);
+                return GetIndexOfFirstNonAsciiByte_Vector(pBuffer, bufferLength);
```

Comment:
nit: leave a comment that it's fine to call this version without any SIMD support?